src/reactor.h

```diff
@@ -0,0 +1,242 @@
+  /// (e.g., reading from the database, instantiating a new object, etc.).
+  /// @warning The Prime Directive must have a space before it! (A fix will be
+  /// in 2.0 ^TM)
+
+  #pragma cyclus
+
+  #pragma cyclus note {"doc": "A stub facility is provided as a skeleton " \
+                              "for the design of new facility agents."}
+
+  /// A verbose printer for the Reactor
+  virtual std::string str();
+
+  /// Set up policies and buffers:
+  virtual void EnterNotify();
+
+  /// The handleTick function specific to the Reactor.
+  /// @param time the time of the tick
+  virtual void Tick();
+
+  /// The handleTick function specific to the Reactor.
+  /// @param time the time of the tock
+  virtual void Tock();
+
+  #pragma cyclus var { \
+    "doc": "Nameplate fusion power of the reactor", \
+    "tooltip": "Nameplate fusion power", \
+    "units": "MW", \
+    "uilabel": "Fusion Power" \
+  }
+  double fusion_power;
+
+  #pragma cyclus var { \
+    "doc": "Achievable system tritium breeding ratio before decay", \
+    "tooltip": "Achievable system tritium breeding ratio before decay", \
+    "units": "non-dimensional", \
+    "uilabel": "Tritium Breeding Ratio" \
+  }
+  double TBR;
+
+  #pragma cyclus var { \
+    "doc": "Minimum tritium inventory to hold in reserve in case of tritium recovery system failure", \
+    "tooltip": "Minimum tritium inventory to hold in reserve (excluding core invneotry)", \
+    "units": "kg", \
+    "uilabel": "Reserve Inventory" \
+  }
+  double reserve_inventory;
+
+  #pragma cyclus var { \
+    "doc": "Tritium core inventory required to start reactor", \
+    "tooltip": "Tritium core inventory required to start reactor", \
+    "units": "kg", \
+    "uilabel": "Start-up Inventory" \
+  }
+  double startup_inventory;
+
+  #pragma cyclus var { \
+    "default": 'fill', \
+    "doc": "Method of refueling the reactor", \
+    "tooltip": "Options: 'schedule' or 'fill'", \
+    "uilabel": "Refuel Mode" \
+  }
+  std::string refuel_mode;
+
+  #pragma cyclus var { \
+    "default": 0.1, \
+    "doc": "Quantity of fuel which reactor tries to purchase", \
+    "tooltip": "Defaults to 100g/purchase", \
+    "units": "kg", \
+    "uilabel": "Buy quantity" \
+  }
+  double buy_quantity;
+
+ #pragma cyclus var { \
+    "default": 1, \
+    "doc": "Frequency which reactor tries to purchase new fuel", \
+    "tooltip": "Reactor is active for 1 timestep, then dormant for buy_frequency-1 timesteps", \
+    "units": "Timesteps", \
+    "uilabel": "Buy frequency" \
+  }
+  double buy_frequency;
+
+  #pragma cyclus var { \
+    "doc": "Fresh fuel commodity", \
+    "tooltip": "Name of fuel commodity requested", \
+    "uilabel": "Fuel input commodity" \
+  }
+  std::string fuel_incommod;
+
+  #pragma cyclus var { \
+    "doc": "Fresh fuel commodity", \
+    "tooltip": "Name of fuel commodity requested", \
+    "uilabel": "Fuel input commodity" \
+  }
+  std::string he3_outcommod;
```

Comment:
This is mainly a product of my early understanding of how cyclus worked. Other agents have a more general input commodity "fuel" which can be things like UOX or MOX, and my thinking was that we wanted the same sort of thing. Currently the code only supports pure Tritium as fuel, and checks to make sure that this is what is supplied before running. However, I think leaving this as something the user can define is better looking towards the future, since even though it currently has to be T, eventually we'll want to be able to run off of a user defined mixture of D and T, but I'm definitely not married to that implementation.